packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -542,6 +543,17 @@ function warnIfInvokedDuringConstruction(vm: VM, methodOrPropName: string) {
         return vm.shadowRoot;
     },
 
+    get elementSelf(): Node | undefined {
+        const vm = getAssociatedVM(this);
+        const isRenderLight = vm.renderMode === RenderMode.Light;
+
+        if (isRenderLight) {
+            return vm.elm;
```

Comment:
I'm pretty sure you can just return `vm.elm` everywhere, without regard to light/shadow mode.